assets/js/modules/analytics-4/datastore/properties.test.js

```diff
@@ -1308,6 +1381,186 @@ describe( 'modules/analytics-4 properties', () => {
 			} );
 		} );
 
+		describe( 'getPropertyCreateTime', () => {
+			it( 'should use a resolver to make a network request', async () => {
+				fetchMock.get( propertyEndpoint, {
+					body: fixtures.properties[ 0 ],
+					status: 200,
+				} );
+
+				const propertyID = '12345';
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.setPropertyID( propertyID );
+
+				const initalPropertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( initalPropertyCreateTime ).toBeUndefined();
+
+				await untilResolved(
+					registry,
+					MODULES_ANALYTICS_4
+				).getPropertyCreateTime();
+				expect( fetchMock ).toHaveFetched( propertyEndpoint, {
+					query: { propertyID },
+				} );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toEqual(
+					new Date( fixtures.properties[ 0 ].createTime ).getTime()
+				);
+				expect( fetchMock ).toHaveFetchedTimes( 1 );
+			} );
+
+			it( 'should reset propertyCreateTime setting when property ID is PROPERTY_CREATE', async () => {
+				provideUserAuthentication( registry );
+
+				fetchMock.get( propertyEndpoint, {
+					body: {},
+					status: 200,
+				} );
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.selectProperty( PROPERTY_CREATE );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBe( 0 );
+
+				expect( fetchMock ).toHaveFetchedTimes( 0 );
+			} );
+
+			it( 'should convert property time to unix timestamp when value is set', () => {
+				const createTimeInDateString = '2014-10-02T15:01:23Z';
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.setPropertyCreateTime( createTimeInDateString );
+
+				const propertyCreateTime = registry
+					.select( MODULES_ANALYTICS_4 )
+					.getPropertyCreateTime();
+
+				expect( propertyCreateTime ).toBe(
+					new Date( createTimeInDateString ).getTime()
+				);
+			} );
+
+			it( 'should reset create time when new property is created', async () => {
```

Comment:
I don't think this is accurate anymore. Originally this branch had changes to the fetch create property action but those were undone. This test is probably passing because the settings are received as empty in the `beforeEach`
https://github.com/google/site-kit-wp/blob/6b4fc66b9c8b574af4e6059d7b3aa9064751ab9b/assets/js/modules/analytics-4/datastore/properties.test.js#L78-L79

`toBeUndefined` is also incorrect as the setting is reset to its default `0` rather than `undefined`.

Reply:
Ah yes indeed, I updated the test to reflect the changes